pkg/kube/util/versionedsecretstore/versioned_secret.go

```diff
@@ -0,0 +1,57 @@
+package versionedsecretstore
+
+import (
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// IsVersionedSecret returns true if the secret has a label identifying it as versioned secret
+func IsVersionedSecret(secret corev1.Secret) bool {
+	secretLabels := secret.GetLabels()
+	if secretLabels == nil {
+		return false
+	}
+
+	if kind, ok := secretLabels[LabelSecretKind]; ok && kind == VersionSecretKind {
+		return true
+	}
+
+	return false
+}
+
+// UnversionedName returns the unversioned name of a secret by removing the /-v\d+/ suffix
+func UnversionedName(name string) (string, error) {
```

Comment:
We can reuse this function: https://github.com/cloudfoundry-incubator/cf-operator/blob/master/pkg/kube/util/names/names.go#L121